components/sync/service/brave_sync_service_impl.cc

```diff
@@ -349,31 +356,48 @@ void BraveSyncServiceImpl::OnSyncCycleCompleted(
 }
 
 void BraveSyncServiceImpl::UpdateP3AObjectsNumber() {
-  GetEntityCountsForDebugging(
-      BindRepeating(&BraveSyncServiceImpl::OnGotEntityCounts,
-                    weak_ptr_factory_.GetWeakPtr()));
+  synced_objects_context_.Reset(GetUserSettings()->GetSelectedTypes().size());
+
+  for (UserSelectableType user_selected_type :
+       GetUserSettings()->GetSelectedTypes()) {
+    ModelType model_type =
+        UserSelectableTypeToCanonicalModelType(user_selected_type);
+
+    ModelTypeController::TypeMap::const_iterator entry =
+        model_type_controllers_.find(model_type);
+    DCHECK(entry != model_type_controllers_.end())
+        << "Missing controller for type " << ModelTypeToDebugString(model_type);
+    if (entry != model_type_controllers_.end()) {
+      const std::unique_ptr<ModelTypeController>& controller = entry->second;
```

Comment:
ModelTypeController* controller = entry->second.get();

Reply:
thanks, fixed that at https://github.com/brave/brave-core/pull/22923/commits/9d0dfb159b6f3adbffda22364c125609dd298cf4